File: android/src/main/java/com/quito/models/Protocol.kt
package com.quito.models

enum class Protocol {
  WS {
    override fun urlPrefix(): String = "ws://"
     },
  WSS {
    override fun urlPrefix(): String = "wss://"
      },
  TCP_TLS {
    override fun urlPrefix(): String = "ssl://"
      },
  TCP {
    override fun urlPrefix(): String = "tcp://"
  };

  abstract fun urlPrefix(): String
}

File: ios/models/Protocol.swift
enum Protocol: String {
  case WS = "WS"
  case WSS = "WSS"
  case TCP_TLS = "TCP_TLS"
  case TCP = "TCP"

  var urlPrefix: String {
      switch self {
        case .WS:
          "ws://"
        case .WSS:
          "wss://"
        case .TCP_TLS:
          "ssl://"
        default:
          "tcp://"
      }
    }
}
